[FLINK-19197] [docs] Add documentation for PersistedStateRegistry (#147)
apache/flink-statefun: docs/sdk/java.md

PersistedAppendingBuffer<Integer> buffer = PersistedAppendingBuffer.of("my-buffer", Integer.class);
{% endhighlight %}

### Dynamic State Registration

Using the above state types, a function's persisted state must be defined eagerly. You cannot use those state types to

> register new persisted state during invocations (i.e. in the ``invoke`` method) or after the function instance is created.

Reviewer: ```suggestion
register a new persisted state during invocations (i.e., in the ``invoke`` method) or after the function instance is created.
```

> If dynamic state registration is required, you can use a ``PersistedStateRegistry`` to achieve that:

Reviewer: I can't exactly explain why, but the way you wrote it sounds off to me. 
```suggestion
If dynamic state registration is required, it can be achieved using a ``PersistedStateRegistry``:
```
